src/Controls/src/Core/HandlerImpl/Application/Application.Impl.cs

```diff
@@ -122,6 +123,11 @@ protected virtual Window CreateWindow(IActivationState? activationState)
 			if (_singleWindowMainPage != null)
 				return new Window(_singleWindowMainPage);
 
+			var windowCreator = activationState?.Context.Services.GetService<IWindowCreator>();
```

Comment:
So, I think we want this block of code to be the first thing that runs inside this method. 

If a user has supplied an `IWindowCreator` then I would equate that to overriding this entire method. 

The reason I think that's important is that users are kind of expected to override this method completely if they are doing multi window.

Line [119](https://github.com/dotnet/maui/pull/14921/files#diff-36e64f9f68575c26cca9f25dc3942affb9b3d3b09bbfae829122b63867d7453bR120-R121) of this method will cause your code to not execute in scenarios where an app is restarting and trying to recreate multiple windows. You can see this happen with Catalyst where if you have multiple windows then destroy the app. If you reopen the app `CreateWindow` will get called for every single window and we want all of those calls to go through your factory. 

@mattleibow